jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java

```diff
@@ -46,4 +46,8 @@ public GenericJBossMarshaller(ClassLoader classLoader, ClassWhiteList classWhite
       baseCfg.setClassResolver(new CheckedClassResolver(classWhiteList, classLoader));
    }
 
+   @Override
+   public void initialize(ClassWhiteList classWhiteList) {
+      baseCfg.setClassResolver(new CheckedClassResolver(classWhiteList, this.getClass().getClassLoader()));
```

Comment:
We have some constructors that take a ClassLoader but I see that classloader is ignored here. Bug ?

Reply:
Yep good spot